lib/airplayer/youtube_dl.rb

```diff
@@ -0,0 +1,92 @@
+require 'shellwords'
+
+module AirPlayer
+  class YoutubeDl
+
+    attr_reader :path
+
+    def initialize(path = self.class.default_path)
+      @path = path
+      # @output = '2> /dev/null'.freeze
+    end
+
+    def enabled?
+      @path && ! @path.empty?
+    end
+
+    EXTRACT_EXTRACTOR_AND_URL = %r{
+      (?<extractor>.+)\n # extractor name and new line
+      \s+(:?.+) # the url if the extractor supports it
+    }x
+
+    GENERIC_EXTRACTOR = 'generic'.freeze
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.